daisy/workflow/validate.go

```diff
@@ -141,6 +142,9 @@ func (w *Workflow) validateRequiredFields() error {
 	if w.Name == "" {
 		return errors.New("must provide workflow field 'Name'")
 	}
+	if !rfc1035Rgx.MatchString(strings.ToLower(w.Name)) {
```

Comment:
I like the idea of moving this type of logic into the steps themselves. right now we kinda have an interface for validate and run, we can add one for populate, although there would be a lot of shared code.